pkg/virt-handler/device-manager/device_controller.go

```diff
@@ -65,8 +65,18 @@ func (c *DeviceController) nodeHasDevice(devicePath string) bool {
 func (c *DeviceController) startDevicePlugin(dev GenericDevice, stop chan struct{}) {
 	logger := log.DefaultLogger()
 	deviceName := dev.GetDeviceName()
+	devicePath := dev.GetDevicePath()
 	retries := 0
 
+	// Making sure the tun and vhost-net modules are loaded on the host
+	// Opening and closing the corresponding dev nodes will trigger modprobes if necessary
+	if deviceName == TunName || deviceName == VhostNetName {
```

Comment:
Would it make sense to add a parameter to `NewGenericDevicePlugin` which indicates that it should pre-open the device? Right now that we have to pre-open is even the "default" case. It might be worth reflecting the need when creating the plugin for the specific device instead of implicit additional checks.

Reply:
Pushed, how does that look?

Reply:
Perfect.